src/components/ReservationForm.tsx

```diff
@@ -0,0 +1,112 @@
+import { useState } from "react";
+import moment from "moment";
+import { Stack, Typography, Box, Button } from "@mui/material";
+
+//THEME
+import { useTheme } from "@mui/material/styles";
+//DATA
+import { bookStayLabel } from "../common/constants";
+//INPUT FIELDS
+import { InputField } from "../components/Form/InputField";
+import { DateField } from "../components/Form/DateField";
+
+export const ReservationForm = () => {
+  const { colors } = useTheme();
+
+  const [formValues, setFormValues] = useState({
+    fullName: "",
+    email: "",
+    numGuests: "",
+    checkIn: "",
+    checkOut: "",
+  });
+
+  const handleChange = (name: string, value: string) => {
+    setFormValues((values) => ({ ...values, [name]: value }));
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    console.log(formValues);
+  };
+
+  const now = moment().format("DD-MM-YYYY");
+  return (
+    <Box>
+      <form onSubmit={handleSubmit}>
+        <Stack spacing={4}>
+          <Typography
+            sx={{
+              color: colors.textBlack,
+              fontFamily: "Roboto",
+              fontWeight: "400",
+              fontSize: "34px",
+            }}
+          >
+            Book your stay
+          </Typography>
+
+          <InputField
+            onChange={handleChange}
+            value={formValues.fullName || ""}
```

Comment:
You don't need to do `var || ""` because you've set all the initial values in `useState` to be an empty string so it is **guaranteed** to be an empty string initially.